app/models/manager_refresh/inventory_collection.rb

```diff
@@ -947,6 +963,19 @@ def full_collection_for_comparison
 
     attr_writer :attributes_blacklist, :attributes_whitelist, :db_data_index, :parent_inventory_collections
 
+    # Returns array of records identities
+    def records_identities(records)
+      records = [records] unless records.kind_of?(Array)
+      records.map { |record| record_identity(record) }
+    end
+
+    # Returns a hash with a simple record identity
+    def record_identity(record)
+      {
+        :id => record.try(:[], :id) || record.try(:id)
```

Comment:
in what scenario could both `try` fail?  if it's possible, we may get one or more {:id => nil} in the arrays, not very useful.  actually `record.id` may also be nil if for some reason it's unsaved.
Can we make it raise exceptions, and elsewhere assume all records in these arrays will have id?